test/test.py

```diff
@@ -1,21 +1,80 @@
-                      Options: ['rf', 'ann', 'lasso', 'xgb']
-        predictors (str): Either a comma-separated list of predictors
-                          or a path to a file with predictor names.
-                          See train/predictors.json for an example.
-
-    Writes performance metrics to data/{SiteID}/{model}/{predictor}/results/
-    for each {SiteID}, {model}, and {predictor} subset.
+        model_dirs (list<str>): Comma-separated list of paths to model
+                                directories with checkpoints. These must all
+                                be from the same site.
+        split (str): Which split to test on. Options: ['train', 'valid', 'test']
+        eval_metrics (list<str>): Metrics to use to evaluate the model(s) on
+                                  the split.
+
+    Writes performance metrics to {model_dir}/{split}_results/
+    for each {model_dir}.
     """
-    pass
\ No newline at end of file
+    if isinstance(model_dirs, str):
+        model_dirs = model_dirs.split(",")
+    splits = ['train', 'valid', 'test']
+    if split not in splits:
+        raise ValueError(f"Got split={split} but must be one of " +
+                         f"{','.join(splits)}")
+
+    site_data_dirs = set(
+        Path(model_dir).parent.parent.parent
+        for model_dir in model_dirs
+    )
+    if len(site_data_dirs) != 1:
+        raise ValueError('Model paths correspond to more than one site.')
+
+    site_data_dir = site_data_dirs.pop()
+
+    eval_scores = []
+    for model_dir in model_dirs:
+        model_dir = Path(model_dir)
+
+        predictor_subset = model_dir.name
+        model = model_dir.parent.name
+
+        if split == 'test':
+            eval_paths = [site_data_dir / 'test.csv']
+            # TODO: Run ensemble model
+        
+        else:
+            # Run each model on the corresponding data split
+            site_training_data_dir = site_data_dir / 'training'
+            n_train = len(list(site_training_data_dir.glob("train*.csv")))
+
+            scores = defaultdict(list)
+            for i in range(n_train):
+                eval_path = site_training_data_dir / f'{split}{i+1}.csv'
+                eval_df = pd.read_csv(eval_path)
+                
+                model_path = model_dir / f"model{i+1}.pkl"
+                if not model_path.exists():
+                    raise ValueError(f"Model path {model_path} does not exist.")
+                with model_path.open('rb') as f:
+                    model_obj = pickle.load(f)
+                
+                for eval_metric in eval_metrics:
+                    score = model_obj.evaluate(eval_df, eval_df['FCH4'], eval_metric)
+                    scores[eval_metric].append(score)
+            
+            scores_df = pd.DataFrame(scores)
+            mean_scores = scores_df.mean().to_frame().T
+            mean_scores['model'] = model
+            mean_scores['predictors'] = predictor_subset
```

Comment:
**Problem**
Unclear presentation of predictors

**Fix**
`mean_scores['predictors'] = predictor_subsets[predictor_subset]`

Reply:
Good point - made a slight modification to this to make sure the "all" predictor setting was expanded.